client/app/react-components/AlertUnsavedChanges.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const unloadMessage = 'You will lose your changes if you leave';
+const confirmMessage = `${unloadMessage}\n\nAre you sure you want to leave this page?`;
+
+export default class AlertUnsavedChanges extends React.Component {
+  static propTypes = {
+    isDirty: PropTypes.bool.isRequired,
+    onChangeLocation: PropTypes.func.isRequired,
+  }
+
+  componentDidMount() {
+    this._onbeforeunload = window.onbeforeunload;
```

Comment:
Interesting trick. I think this could've helped me in clearing a debounced event listener once.